### PR TITLE
fix(socket sink): Emit warning on incomplete UDP sent

### PR DESCRIPTION
**Pull request by juchiast**
Signed-off-by: Duy Do <juchiast@gmail.com>

### CODE REVIEW

File: src/sinks/util/udp.rs [hunk 1]
```diff
@@ -171,7 +171,16 @@ impl Sink for UdpSink {
                         error!(message = "send failed", %error);
                         Ok(AsyncSink::NotReady(line))
                     }
-                    Ok(_) => Ok(AsyncSink::Ready),
+                    Ok(sent) => {
+                        if sent != line.len() {
+                            warn!(
+                                "Could not send all data in one UDP packet, data length: {}, sent: {}.",
```
Comment (ktff) [inline src/sinks/util/udp.rs:177]: Extending the message to convey why is this important would also be a nice addition, something like `; dropping X bytes.`

Comment (juchiast) [reply src/sinks/util/udp.rs]: Thanks! Added that.

**Review by ktff — APPROVED** [review]

File: src/sinks/util/udp.rs [hunk 2]
```diff
@@ -171,7 +171,17 @@ impl Sink for UdpSink {
                         error!(message = "send failed", %error);
                         Ok(AsyncSink::NotReady(line))
                     }
-                    Ok(_) => Ok(AsyncSink::Ready),
+                    Ok(sent) => {
+                        if sent != line.len() {
+                            warn!(
```
Comment (binarylogic) [inline src/sinks/util/udp.rs:176]: @juchiast is this better represented as an event? And should it the log be rate-limited?

Comment (juchiast) [reply src/sinks/util/udp.rs]: Good point! We don't have back-off for this event so it is better to rate-limit it. I added an internal event for this. I also changed it to `error!`, not sure why I used `warn!` originally.

**Review by binarylogic — APPROVED** [review]

**Pull request merged**